forward/ft_inside_vol.m

```diff
@@ -120,59 +120,74 @@
-    % determine all vertices that are part of a brain hexaheder
-    insidepos = false(numpos,1);
-    insidepos(headmodel.hex(insidehex,:)) = true;
-    
+    % construct a sparse matrix with the mapping between all hexaheders and vertices
     i = repmat(transpose(1:numhex), 1, 8);
     j = headmodel.hex;
     s = ones(size(i));
     hex2pos = sparse(i(:),j(:),s(:),numhex,numpos);
     
+    % determine all vertices that are part of a brain hexaheder
+    insidepos = false(numpos,1);
+    insidepos(headmodel.hex(insidehex,:)) = true;
     % prune the mesh, i.e. only retain vertices that are part of a brain hexaheder
-    fullindx = find(insidepos);
-    fprintf('pruning mesh from %d to %d vertices (%d%%)\n', numel(insidepos), sum(insidepos), round(100*sum(insidepos)/numel(insidepos)));
-    meshpos = headmodel.pos(fullindx,:);
+    fprintf('pruning headmodel vertices from %d to %d (%d%%)\n', numpos, sum(insidepos), round(100*sum(insidepos)/numpos));
+    insidepos  = find( insidepos);
+    meshpos = headmodel.pos(insidepos,:);
+
+    % determine the bounding box
+    minpos = min(meshpos,[],1);
+    maxpos = max(meshpos,[],1);
+    insidedip = all(bsxfun(@ge, dippos, minpos),2) & all(bsxfun(@le, dippos, maxpos),2);
+    fprintf('pruning dipole positions from %d to %d (%d%%)\n', numdip, sum(insidedip), round(100*sum(insidedip)/numdip));
+    insidedip  = find( insidedip);
+    dippos = dippos(insidedip,:);
     
-    % find the nearest vertex for each of the dipole positions
+    % find the nearest vertex for each of the dipoles
     dsearchn(meshpos, dippos(1,:)); % call it once to precompile
     stopwatch = tic;
-    dsearchn(meshpos, dippos(1,:)); % call it once to determine the time
+    dsearchn(meshpos, dippos(1,:)); % call it once to estimate the time
     t = toc(stopwatch);
-    fprintf('determining inside points, this takes about %d seconds\n', round(size(dippos,1)*t));
+    fprintf('determining inside points, this takes about %d seconds\n', round(numdip*t));
     subindx = dsearchn(meshpos, dippos);
     
     % The following code is only guaranteed to work with convex elements. Regular
     % hexahedra and tetrahedra are convex, and the adapted hexahedra we can use with
-    % SIMBIO have to be convex.
+    % SIMBIO have to be convex as well.
     
     inside = false(1, numdip);
     % for each dipole determine whether it is inside one of the neighbouring hexaheders
     % this will be the case for all vertices that are inside the middle, but not at the edges
-    for i=1:numdip
-      hexindx = find(hex2pos(:,fullindx(subindx(i))));
+    for i=1:numel(insidedip)
+      hexindx = find(hex2pos(:,insidepos(subindx(i))));
       for j=1:numel(hexindx)
         posx = headmodel.pos(headmodel.hex(hexindx(j),:),1);
         posy = headmodel.pos(headmodel.hex(hexindx(j),:),2);
         posz = headmodel.pos(headmodel.hex(hexindx(j),:),3);
         pos = [posx posy posz];
         if isequal(convhull(pos), convhull([pos; dippos(i,:)]))
```

Comment:
The problem occurs if a source falls into an non-brain element that have a sharing face with a brain element.  I.e. A source point is inside such a non-brain element. We found the nearest neighbouring vertices and look up all elements that share this vertex. After that we only checked if it inside one of this 8 element not if this element is part of the brain.